src/models.py

```diff
@@ -931,6 +957,42 @@ def _lm_forward(self, batch, task, predict):
             pass
         return out
 
+    def _lm_only_forward(self, batch, task, predict):
+        """Only forward pass for LM model - consider for non-bidirectional models
+        Args:
+            batch: indexed input data
+            task: (Task obejct)
+            predict: (boolean) predict mode (not supported)
+        return:
+            out: (dict)
+                - 'logits': output layer, dimension: [batchSize * timeSteps * 2, outputDim]
+                            first half: [:batchSize*timeSteps, outputDim] is output layer from forward layer
+                            second half: [batchSize*timeSteps:, outputDim] is output layer from backward layer
+                - 'loss': size average CE loss
+        """
+        out = {}
```

Comment:
If this is your code, add some comments and newlines to make it clear what the high-level steps are.

Reply:
(Newlines would still help.)

Reply:
Added a newline before the code. Some comments explain what the code does.

Reply:
Take another look at the rest of the file. This is workable as-is, but still harder to read than it should be.